flink-runtime/src/main/java/org/apache/flink/runtime/memory/UnsafeMemoryBudget.java

```diff
@@ -147,8 +147,10 @@ void reserveMemory(long size, int maxSleeps) throws MemoryReservationException {
 			}
 
 			// no luck
-			throw new MemoryReservationException(
-				String.format("Could not allocate %d bytes, only %d bytes are remaining", size, availableOrReserved));
+			throw new MemoryReservationException(String.format(
+				"Could not allocate %d bytes, only %d bytes are remaining, try to upgrade to Java 8u72 or higher",
```

Comment:
The warning can be overlooked. We agreed with @tillrohrmann that we want to duplicate it here if the warning has been overlooked.

Reply:
Yes adding a bit more context for why to upgrade would be helpful. E.g. "This usually indicates that you are requesting more memory than you have reserved. However, when running an old JVM version it can also be caused by slow garbage collection. Try to upgrade to Java 8u72 or higher if running on an old Java version.".